nixos/modules/services/hardware/xow.nix

```diff
@@ -0,0 +1,46 @@
+{ config, pkgs, lib, ... }:
+
+let
+  cfg = config.services.xow;
+in {
+  options.services.xow = {
+    enable = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = ''
+        Whether to enable xow or not.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    users.users.uinput = {
+      group = "uinput";
+      isSystemUser = true;
+    };
+
+    users.groups.uinput = {};
+
+    services.udev.packages = lib.singleton (pkgs.writeTextFile {
+      name = "xow-uinput-rules";
+      destination = "/etc/udev/rules.d/99-uinput.rules";
+      text = ''
+        SUBSYSTEM=="misc", KERNEL=="uinput", MODE="0660", GROUP="uinput", OPTIONS+="static_node=uinput"
+      '';
+    });
+
+    environment.systemPackages = [ pkgs.xow ];
```

Comment:
Unless the user needs to have this in their environment to call a binary themselves or so, this can be removed.

Reply:
Fixed.